docs/scripting.md

```diff
@@ -698,24 +716,50 @@ module.exports = (robot) ->
 
 ## Listener Middleware API
 
-Although internal data structures are exposed, not all properties on the objects are considered part of the supported API. Below are the supported properties and usage information for each of the arguments middleware receives.
-
-- `context`
+Listener middleware callbacks receive three arguments, `context`, `next`, and
+`done`. See the [middleware API](#execution-process-and-api) for a description
+of `next` and `done`. Listener middleware context includes these fields:
   - `listener`
     - `options`: a simple Object containing options set when defining the listener. See [Listener Metadata](#listener-metadata).
     - all other properties should be considered internal
-
   - `response`
     - all parts of the standard response API are included in the middleware API. See [Send & Reply](#send--reply).
     - middleware may decorate (but not modify) the response object with additional information (e.g. add a property to `response.message.user` with a user's LDAP groups)
     - note: the textual message (`response.message.text`) should be considered immutable in listener middleware
 
-- `next`
-  - a Function with no additional properties that should be called to continue on to the next piece of middleware/execute the Listener callback
-  - `next` should be called with a single optional argument: either the provided `done` function or a new function that eventually calls `done`
+# Receive Middleware
 
-- `done`
- - a Function with no additional properties that should be called to interrupt middleware execution and begin executing the chain of completion functions.
- - `done` should be called with no arguments
+Receive middleware runs before any listeners have executed. It's suitable for
+blacklisting commands that have not been updated to add an ID, metrics, and more.
+
+## Receive Middleware Example
+
+This simple middlware bans hubot use by a particular user, including `hear`
```

Comment:
I'm confused. Do you mean `robot.hear /^hubot: stuff$/` would get ignored? Yes, I agree, but that isn't how the docs read. I would interpret the docs as "This will block `robot.hear /hello/`", but that is false.